documentation/docs/02-layouts.md

```diff
@@ -62,11 +62,75 @@ We can create a layout that only applies to pages below `/settings` (while inher
 <slot></slot>
 ```
 
-### Resets
+### Named layouts
 
-To reset the layout stack, create a `__layout.reset.svelte` file instead of a `__layout.svelte` file. For example, if you want your `/admin/*` pages to _not_ inherit the root layout, create a file called `src/routes/admin/__layout.reset.svelte`.
+Some parts of your app might need something other than the default layout. For these cases you can create _named layouts_...
 
-Layout resets are otherwise identical to normal layout components.
+```svelte
+/// file: src/routes/__layout-foo.svelte
+<div class="foo">
+	<slot></slot>
+</div>
+```
+
+...and then use them by referencing the layout name (`foo`, in the example above) in the filename:
+
+```svelte
+/// file: src/routes/my-special-page@foo.svelte
+<h1>I am inside __layout-foo</h1>
+```
+
+Named layouts are very powerful, but it can take a minute to get your head round them. Don't worry if this doesn't make sense all at once.
+
+#### Scoping
+
+Named layouts can be created at any depth, and will apply to any components in the same subtree. For example, `__layout-foo` will apply to `one` and `two`, but not `three` or `four`:
+
+```
+src/routes/
+├ a/
+│ ├ b/
+│ │ ├ c/
+│ │ │ ├ __layout-foo.svelte
+│ │ │ ├ one@foo.svelte
+│ │ │ ├ two@foo.svelte
+│ │ │ └ three.svelte
+│ │ └ four@foo.svelte
+```
+
+#### Inheritance chains
+
+Layouts can themselves choose which parent layouts they inherit from. Ordinarily, they will inherit from any default layouts 'above' them in the tree, but they can inherit named layouts using the same logic as pages. For example, `a/b/c/__layout@foo.svelte` is the default layout for `a/b/c`, because it has no name (meaning `one`, `two` and `three` all inherit from it), and because it specifies `@foo`, it will inherit directly from the nearest `__layout-foo.svelte`, skipping `__layout.svelte`, `a/__layout.svelte` and `a/b/__layout.svelte`.
+
+```
+src/routes/
+├ a/
+│ ├ b/
+│ │ ├ c/
+│ │ │ ├ __layout@foo.svelte
+│ │ │ ├ one.svelte
+│ │ │ ├ two.svelte
+│ │ │ └ three.svelte
+│ │ └ __layout.svelte
+│ └ __layout.svelte
+├ __layout.svelte
+└ __layout-foo.svelte
+```
+
+Layouts that specify a parent will ordinarily ignore any default layouts, but you can force a named layout to inherit from the nearest default layout with the reserved `default` parent identifier. For example, a layout called `__layout-home@default.svelte` that contained a single `<slot/>` in the root directory would effectively enable any page, anywhere in the app, to inherit directly from the root layout, ignoring any intermediate layouts:
+
+```
+src/routes/
+├ a/
+│ ├ b/
+│ │ ├ c/
+│ │ │ ├ __layout.svelte                     # skipped
+│ │ │ └ page-with-root-layout@home.svelte
+│ │ └ __layout.svelte                       # skipped
+│ └ __layout.svelte                         # skipped
+├ __layout.svelte                           # inherited
+└ __layout-home@default.svelte              # inherited
+```
 
```

Comment:
Not great englishing on my part, but this is probably a good place to take resets home (and make them searchable):

````suggestion
#### Layout resets

Taking into account scoping and inheritance rules, we can reset layouts to any depth we wish by carefully creating named layouts at the right position in the tree:

```
src/routes/
├ __layout.svelte
├ __layout-blank.svelte
└ a/
  ├ __layout@a.svelte
  ├ __layout-a.svelte
  └ b/
    ├ __layout.svelte
    └ c/
      ├ __layout.svelte
      ├ page-with-c's-layout.svelte
      ├ page-with-a's-layout@a.svelte
      └ page-with-empty-layout@blank.svelte
```

````

Reply:
I've been using this ropey-as-hell REPL app to generate the file tree layouts: https://svelte.dev/repl/5f66998f2c924a9eadb0954896682616?version=3.46.6. To make it consistent with the others it'd look more like this:


````suggestion
#### Layout resets

Taking into account scoping and inheritance rules, we can reset layouts to any depth we wish by carefully creating named layouts at the right position in the tree:

```
src/routes/
├ a/
│ ├ b/
│ │ ├ c/
│ │ │ ├ __layout.svelte
│ │ │ ├ page-with-c's-layout.svelte
│ │ │ ├ page-with-a's-layout@a.svelte
│ │ │ └ page-with-empty-layout@blank.svelte
│ │ └ __layout.svelte
│ ├ __layout@a.svelte
│ └ __layout-a.svelte
├ __layout.svelte
└ __layout-blank.svelte
```

````

Though I wonder if the `__layout-a`/`__layout@a` pair is a bit cryptic as things stand. Maybe something like this would be easier to follow? Though it uses a slightly different sorting to the other blocks, so they'd probably need to change. 

```
src/routes/
├ __layout.svelte        # (1)
├ __layout-root.svelte   # (2)
├ foo/
│ ├ __layout-foo.svelte  # (3)
│ ├ __layout@foo.svelte  # (4)
│ ├ bar/
│ │ ├ __layout.svelte    # (5)
│ │ ├ one.svelte         # 1 -> 3 -> 4 -> 5 -> one.svelte
│ │ ├ two@foo.svelte     # 1 -> 3 -> 4 -> two@foo.svelte
│ │ └ three@root.svelte  # 2 -> three@root.svelte
│ └ four.svelte          # 1 -> 3 -> 4 -> four.svelte
```

````

Reply:
Yeah, and that's a use-case I would expect most people would never touch, so it was conflicting adding it.

Would inverting it to `__layout-foo@default.svelte` be easier or hard to grasp, I wonder? Or perhaps moving it a layer down like `/a/b/__layout-extends-a.svelte` (used `a` to be referenced at `c` since `/foo/bar` wouldn't be deep enough to get the point across)?

Reply:
I started writing "I think inverting it probably makes a ton of sense, since `@default` has already been discussed"...

````suggestion
#### Layout resets

Taking into account scoping and inheritance rules, we can reset layouts to any depth we wish by carefully creating named layouts at the right position in the tree:

```
src/routes/
├ __layout.svelte                # (1)
├ __layout-root.svelte           # (2)
├ foo/
│ ├ __layout.svelte              # (3)
│ ├ __layout-foo@default.svelte  # (4)
│ ├ bar/
│ │ ├ __layout.svelte            # (5)
│ │ ├ one.svelte                 # 1 -> 3 -> 5 -> one.svelte
│ │ ├ two@foo.svelte             # 1 -> 3 -> 4 -> two@foo.svelte
│ │ └ three@root.svelte          # 2 -> three@root.svelte
│ └ four.svelte                  # 1 -> 3 -> four.svelte
```

````

...but then we miss the fact that `four.svelte` inherits from a layout that inherits from the `foo` layout. Gah...

Reply:
I'm not sure I follow—the example is supposed to be about resetting to layouts at arbitrary depths, and `__layout-foo@default` would indeed be an empty `<slot/>` anchor into `/foo/__layout`. Or at least that was my original intention with the "using `a`'s layout" example.

Reply:
Although, yeah, `/foo/bar` isn't deep enough since `/foo/bar/__layout-alt.svelte` would have the same meaning.

It'd make more sense having `/foo/__layout-foo@default.svelte` and `/foo/bar/baz/page@foo.svelte` to hit the idea that `page` shares the same layout as `/foo/four.svelte`.

Reply:
just gonna go with the most recent suggestion for now

Reply:
oh hang on, missed your comments because github's UI sucks balls